File: src/main/java/catmoe/fallencrystal/akanelimbo/serverlist/ServerListCommand.java
package catmoe.fallencrystal.akanelimbo.serverlist;

import java.util.List;
import java.util.Map;

import catmoe.fallencrystal.akanelimbo.command.SubCommand;
import net.md_5.bungee.api.CommandSender;
import net.md_5.bungee.api.connection.ProxiedPlayer;

public class ServerListCommand implements SubCommand {

    @Override
    public String getSubCommandId() {
        return "serverlist";
    }

    @Override
    public void execute(CommandSender sender, String[] args) {
        ServerListMenu menu = new ServerListMenu();
        menu.open((ProxiedPlayer) sender);
    }

    @Override
    public String getPermission() {
        return "bungeecord.command.server";
    }

    @Override
    public Map<Integer, List<String>> getTabCompleter() {
        return null;
    }

    @Override
    public boolean allowedConsole() {
        return false;
    }

    @Override
    public boolean StrictSizeLimit() {
        return true;
    }

    @Override
    public int StrictSize() {
        return 1;
    }

}

File: src/main/java/catmoe/fallencrystal/akanelimbo/serverlist/ServerListCommand.kt
package catmoe.fallencrystal.akanelimbo.serverlist

import catmoe.fallencrystal.akanelimbo.StringManager
import catmoe.fallencrystal.akanelimbo.command.SubCommand
import net.md_5.bungee.api.CommandSender
import net.md_5.bungee.api.connection.ProxiedPlayer

class ServerListCommand : SubCommand {

    override val subCommandId: String get() = "serverlist"

    override fun execute(sender: CommandSender?, args: Array<String>) {
        val menu = ServerListMenu()
        menu.open((sender as ProxiedPlayer))
    }

    private fun getTabCompleterMap(): MutableMap<Int, List<String>> {
        val map : MutableMap<Int, List<String>> = HashMap()
        val tip1 = ArrayList<String>()
        tip1.add("?")
        map[1] = tip1
        return map
    }

    override val permission: String get() = StringManager.getServerListPermission()
    override val tabCompleter: MutableMap<Int, List<String>> get() = getTabCompleterMap()
    override fun allowedConsole(): Boolean { return false }
    override fun strictSizeLimit(): Boolean { return true }
    override fun strictSize(): Int { return 1 }
}
